handsontable/src/plugins/formulas/formulas.js

```diff
@@ -183,26 +201,64 @@ export class Formulas extends BasePlugin {
     this.addHook('afterRemoveRow', (...args) => this.onAfterRemoveRow(...args));
     this.addHook('afterRemoveCol', (...args) => this.onAfterRemoveCol(...args));
 
+    this.indexSyncer = new IndexSyncer(this.hot.rowIndexMapper, this.hot.columnIndexMapper, (postponedAction) => {
+      this.hot.addHookOnce('init', () => {
+        // Engine is initialized after executing callback to `afterLoadData` hook. Thus, some actions on indexes should
+        // be postponed.
+        postponedAction();
+      });
+    });
+
+    this.rowAxisSyncer = this.indexSyncer.getForAxis('row');
+    this.columnAxisSyncer = this.indexSyncer.getForAxis('column');
+
+    this.hot.addHook('afterRowSequenceChange', this.rowAxisSyncer.getIndexesChangeSyncMethod());
+    this.hot.addHook('afterColumnSequenceChange', this.columnAxisSyncer.getIndexesChangeSyncMethod());
+
+    this.hot.addHook('beforeRowMove', this.rowAxisSyncer.getBeforeMoveMethod());
+    this.hot.addHook('beforeColumnMove', this.columnAxisSyncer.getBeforeMoveMethod());
+
+    this.hot.addHook('afterRowMove', this.rowAxisSyncer.getIndexMoveSyncMethod());
+    this.hot.addHook('afterColumnMove', this.columnAxisSyncer.getIndexMoveSyncMethod());
+
+    this.hot.addHook('beforeColumnFreeze', (column) => {
+      this.columnAxisSyncer.getBeforeMoveMethod()([column], this.hot.getSettings().fixedColumnsStart);
+    });
+
+    this.hot.addHook('afterColumnFreeze', (_, freezePerformed) => {
+      this.columnAxisSyncer.getIndexMoveSyncMethod()(undefined, undefined, undefined, freezePerformed);
```

Comment:
Weird API implementation that requires passing `undefined, undefined, undefined...` at the beginning of the method 🤔

Reply:
Changed within https://github.com/handsontable/handsontable/pull/10215/commits/f73286fcfae24866906e7e8ce02256dd1aa4e47e.